pkg/apiserver/decisions_test.go

```diff
@@ -251,181 +257,902 @@ func TestStreamStartDecisionDedup(t *testing.T) {
+			Method:        "GET",
+			Route:         "/v1/decisions/stream?startup=true&scenarios_not_containing=ssh_bf,test",
+			CheckCodeOnly: false,
+			Code:          200,
+			LenNew:        1,
+			LenDeleted:    0,
+			AuthType:      APIKEY,
+			DelChecks:     []DecisionCheck{},
+
+			NewChecks: []DecisionCheck{
+				{
+					ID:       int64(3),
+					Origin:   "test",
+					Scenario: "crowdsecurity/longest",
+					Value:    "127.0.0.1",
+					Duration: "4h59",
+				},
+			},
+		},
+		{
+			TestName:      "test startup with origins parameter",
+			Method:        "GET",
+			Route:         "/v1/decisions/stream?startup=true&origins=another_origin",
+			CheckCodeOnly: false,
+			Code:          200,
+			LenNew:        2,
+			LenDeleted:    0,
+			AuthType:      APIKEY,
+			DelChecks:     []DecisionCheck{},
+
+			NewChecks: []DecisionCheck{
+				{
+					ID:       int64(7),
+					Origin:   "another_origin",
+					Scenario: "crowdsecurity/test",
+					Value:    "127.0.0.2",
+					Duration: "1h59",
+				},
+				{
+					ID:       int64(2),
+					Origin:   "another_origin",
+					Scenario: "crowdsecurity/ssh_bf",
+					Value:    "127.0.0.1",
+					Duration: "2h59",
+				},
+			},
+		},
+		{
+			TestName:      "test startup with multiple origins parameter",
+			Method:        "GET",
+			Route:         "/v1/decisions/stream?startup=true&origins=another_origin,test",
+			CheckCodeOnly: false,
+			Code:          200,
+			LenNew:        2,
+			LenDeleted:    0,
+			AuthType:      APIKEY,
+			DelChecks:     []DecisionCheck{},
+
+			NewChecks: []DecisionCheck{
+				{
+					ID:       int64(4),
+					Origin:   "test",
+					Scenario: "crowdsecurity/test",
+					Value:    "127.0.0.2",
+					Duration: "2h59",
+				},
+				{
+					ID:       int64(3),
+					Origin:   "test",
+					Scenario: "crowdsecurity/longest",
+					Value:    "127.0.0.1",
+					Duration: "4h59",
+				},
+			},
+		},
+		{
+			TestName:      "test startup with unknown origins",
+			Method:        "GET",
+			Route:         "/v1/decisions/stream?startup=true&origins=unknown",
+			CheckCodeOnly: false,
+			Code:          200,
+			LenNew:        0,
+			LenDeleted:    0,
+			AuthType:      APIKEY,
+			DelChecks:     []DecisionCheck{},
+			NewChecks:     []DecisionCheck{},
+		},
+		{
+			TestName:      "delete decisions 3 (127.0.0.1)",
+			Method:        "DELETE",
+			Route:         "/v1/decisions/3",
+			CheckCodeOnly: true,
+			Code:          200,
+			LenNew:        0,
+			LenDeleted:    0,
+			AuthType:      PASSWORD,
+			DelChecks:     []DecisionCheck{},
+			NewChecks:     []DecisionCheck{},
+		},
+		{
+			TestName:      "check that 127.0.0.1 is not in deleted IP",
+			Method:        "GET",
+			Route:         "/v1/decisions/stream?startup=true",
+			CheckCodeOnly: false,
+			Code:          200,
+			LenNew:        2,
+			LenDeleted:    0,
+			AuthType:      APIKEY,
+			DelChecks:     []DecisionCheck{},
+			NewChecks:     []DecisionCheck{},
```

Comment:
I think we should check that we get the expected decisions (duration + scenario)